### PR TITLE
Release/8.1.0

### PR DESCRIPTION
**Pull request by southbite**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -5,9 +5,9 @@ branches:
   only:
     - master
 node_js:
+  - "10"
   - "8"
   - "7"
```
Comment (JohanvdWest) [inline .travis.yml]: Remove node 7 from testing. We can add 11.

File: lib/client.js [hunk 2]
```diff
@@ -816,18 +816,19 @@
 
     //write our session cookie
     if (browser) {
+
       var cookie = (result.cookieName || 'happn_token') + '=' + this.session.token + '; path=/;';
+      if (this.options.protocol == 'https') cookie += ' Secure;';
       if (result.cookieDomain) cookie += ' domain=' + result.cookieDomain + ';';
+      if (this.options.protocol == 'https') cookie += ' Secure;';
```
Comment (JensEggers) [inline lib/client.js]: line 823 appears to repeat 821

Comment (JensEggers) [reply lib/client.js]: ah it's 2 cookies

File: lib/services/security/service.js [hunk 3]
```diff
@@ -1330,12 +1341,18 @@ function login(credentials, sessionId, callback) {
   if (!((credentials.username && (credentials.password || credentials.digest)) || credentials.token))
     return callback(this.happn.services.error.InvalidCredentialsError('Invalid credentials'));
 
+  if (credentials.username == '_ADMIN' && this.config.disableDefaultAdminNetworkConnections && request.data.info._local == false)
```
Comment (JohanvdWest) [inline lib/services/security/service.js]: Who sets the _local?

File: test/browser/browser_client.js [hunk 4]
```diff
@@ -1 +1 @@
...er(t,e)}),t.prototype.offPath=c(function(t,e){return"function"==typeof t&&(e=t,t="*"),this._offPath(t,e)}),t.prototype.clearTimeouts=function(){var t=this;Object.keys(t.state.ackHandlers).forEach(function(e){clearTimeout(t.state.ackHandlers[e].timedout)}),Object.keys(t.state.requestEvents).forEach(function(e){clearTimeout(t.state.requestEvents[e].timedout)})},t.prototype.__connectionCleanup=function(t){try{if(this.socket){var e=this;if(t&&t.revokeSession)return e.revokeSession(function(t){t&&e.log.warn("socket.end failed in client, revoke session failed: "+t.toString()),e.socket.removeAllListeners("end"),e.socket.end()});this.socket.removeAllListeners("end"),this.socket.end()}}catch(n){this.log.warn("socket.end failed in client: "+n.toString())}},t.prototype.revokeSession=c(function(t){this.__performSystemRequest("revoke-session",null,null,t)}),t.prototype.disconnect=c(function(t,e){try{"function"==typeof t&&(e=t,t=null),t||(t={}),e||(e=function(){}),this.clearTimeouts(),this.__connectionCleanup(t),this.state.systemMessageHandlers=[],this.state.eventHandlers={},this.state.events={},this.state.refCount={},this.state.listenerRefs={},this.status=i.DISCONNECTED,e()}catch(n){e(n)}})}();
```
Comment (JohanvdWest) [inline test/browser/browser_client.js]: Should this file be checked in, or generated if there are changes to client.js?

File: test/browser/01_security_hsts_cookie.js [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+        socketClientHttp = instance;
+        callback();
+      });
+    } catch (e) {
+      callback(e);
+    }
+  });
+
+  it('checks our cookie is not set to secure on socket login', function (done) {
+    if (!document.cookie) return done(new Error('cookie should not be blank'));
```
Comment (JensEggers) [inline test/browser/01_security_hsts_cookie.js]: this doesn't seem to check the secure setting

File: test/integration/security/admin_on_intra_proc_only.js [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+      }, function (e, instance) {
+
+        expect(e.toString()).to.be('AccessDenied: use of _ADMIN credentials over the network is disabled');
+        done();
+      });
+  });
+
+  it('fails to authenticate with the _ADMIN user, on the locked service, using a session token', function (done) {
+
+    serviceInstanceLocked.services.session.localClient({
```
Comment (JohanvdWest) [inline test/integration/security/admin_on_intra_proc_only.js]: I can't find the code that sets localClient on the session.

File: test/browser/browser_client.js (hunk 4 above)
Comment (JohanvdWest) [reply test/browser/browser_client.js]: Sorry, see it's being deleted.

**Pull request merged**